tests/build.rs

```diff
@@ -1789,13 +1789,15 @@ fn example_bin_same_name() {
         .unwrap();
 
     assert_that(&p.bin("foo"), is_not(existing_file()));
+    // We expect a file of the form bin/foo-{metadata_hash}
     assert_that(&p.bin("examples/foo"), existing_file());
 
     p.cargo("test").arg("--no-run").arg("-v")
                    .exec_with_output()
                    .unwrap();
 
     assert_that(&p.bin("foo"), is_not(existing_file()));
+    // We expect a file of the form bin/foo-{metadata_hash}
```

Comment:
Nope this is intentional. `bin/foo` does not exist, but `bin/foo-{metadatahash}` and `bin/examples/foo` both exist. The test is making sure that both files are created, but it can't test for `bin/foo-{metadata hash}` effectively.